addon/serializers/application.js

```diff
@@ -190,7 +191,7 @@ export default DS.RESTSerializer.extend({
       hasMany.forEach( function( child ) {
         json[key].objects.push({
           '__type'    : 'Pointer',
-          'className' : child.parseClassName(),
+          'className' : serializer.parseClassName( child.typeKey ),
```

Comment:
I think it may be more appropriate to use the `serializer` of the `child` type instead?